### DISCUSSION TITLE
cat functions in MultipleTables._summaryPrint()

### DISCUSSION THREAD

**Discussion by esuxiao** [esuxiao/mmeta#12] [Q&A] [answered]
Here the review comments  from CRAN: 

> You write information messages to the console that cannot be easily suppressed. It is more R like to generate objects that can be used to extract the information a user is interested in, and then print() that object.
Instead of print()/cat() rather use message()/warning()  or
if(verbose)cat(..) (or maybe stop()) if you really have to write text to the console.
(except for print, summary, interactive functions) 

But in MultipleTables._summaryPrint function, I still use cat() function to print out information in console. 
https://github.com/esuxiao/mmeta/blob/3dad01049d1ae980a0213a0f1a3e7ae7572f4253/R/MultipleTables-method-internal.R#L345

This  MultipleTables._summaryPrint() is called in a summary method when verbose is True:
https://github.com/esuxiao/mmeta/blob/3dad01049d1ae980a0213a0f1a3e7ae7572f4253/R/MultipleTables-method-exported.R#L108

Thus I think it satisfies the following criteria:

> if(verbose)cat(..) (or maybe stop()) if you really have to write text to the console.(except for print, summary, interactive functions)

**Comment by jchen1992** [comment, marked as answer]
Thanks. Make sense.

**Comment by esuxiao** [comment]
@jchen1992 repose

**Reply by jchen1992** [reply]
https://github.com/esuxiao/mmeta/issues/21#issue-1522983393

Reposed and closed.
